DataFormats/Candidate/interface/VertexCompositePtrCandidate.h

```diff
@@ -15,63 +15,97 @@
 namespace reco {
   class VertexCompositePtrCandidate : public CompositePtrCandidate {
   public:
+    enum { dimension4D = 4 };
+    /// covariance error matrix (3x3)
+    typedef math::Error<dimension4D>::type CovarianceMatrix4D;
+    /// matix size
+    enum { size4D = dimension4D * (dimension4D + 1)/2 };
+
     VertexCompositePtrCandidate() : CompositePtrCandidate() { }
     /// constructor from values
```

Comment:
please initialize data members in the default constructor.
There is an issue reading older version data: time_ comes out with random values.
Instead of setting it to 0 with ioread rules I think it's better to set defaults in the constructor.